modules/tinymce/src/plugins/link/main/ts/core/NormalizeLink.ts

```diff
@@ -0,0 +1,43 @@
+import DomTreeWalker from 'tinymce/core/api/dom/TreeWalker';
+import Editor from 'tinymce/core/api/Editor';
+
+const isBR = (elm: Node): elm is HTMLAnchorElement => elm && elm.nodeName.toLowerCase() === 'br';
+const isAnchor = (elm: Node): elm is HTMLAnchorElement => elm && elm.nodeName.toLowerCase() === 'a';
+
+const getSelectedLink = (editor: Editor, selectedElm?: Element): HTMLAnchorElement | null => {
+  const isShortEnded = (elm: Node) => elm && editor.schema.getShortEndedElements().hasOwnProperty(elm.nodeName);
+  const isSameSiblingAnchor = (elm: Node, sibling: Node): elm is HTMLAnchorElement => (
+    elm === sibling.parentNode ||
+    sibling.parentNode && elm === sibling.parentNode.nextSibling
+  );
+  const isValidNode = (node: Node) => !isShortEnded(node) && isAnchor(node);
+
+  const rng = editor.selection.getRng();
+  const start = rng.startContainer;
+  const end = rng.endContainer;
+
+  const endOffsetChild = end.childNodes[rng.endOffset];
+  const startOffsetChild = start.childNodes[rng.startOffset];
+
+  if (end.nodeType === 1 && isBR(endOffsetChild)) {
+    const walker = new DomTreeWalker(endOffsetChild, endOffsetChild.parentNode);
+
+    while (walker.current()) {
+      const node = walker.prev();
+      if (isValidNode(node) && isSameSiblingAnchor(node, start)) {
```

Comment:
I think this could instead be a simple eq if you took the start node and did a `editor.dom.getParent(selectedElm, 'a[href]')
` on that. Maybe pulling that out into a `getClosestLink` function or something. So in the if statement you get the link element of the start node then in the while you do `if (isValidNode(node) && node === startLink) {`.

Reply:
updated